### PR TITLE
feat(wallet)_: Add PDF viewer screen to view ParaSwap TnC

### LINKED ISSUE #21012: Create embedded webview for Paraswap TOC on mobile
Add it here 
https://www.figma.com/design/AD2JSKg0I8dZcylyiGa62O/Swap-for-Mobile?node-id=860-41069&t=rjgpkkg14jGvc89b-1

### LINKED ISSUE #21427: 🔁 Update swap final screen to show Terms of Services below the slider
# Feature Issue

### User Story

As a user, 
I want to know my rights, and agreements with the swap provider (Paraswap),
So that I can take informed decisions

### Description

We need to make the Paraswap ToS more explicit in the Swap Summary screen (The screen where the users moves the slider to sign and enters her password).

This updated design for the view is on Figma: https://www.figma.com/design/AD2JSKg0I8dZcylyiGa62O/Swap-for-Mobile?node-id=860-41069&t=rjgpkkg14jGvc89b-1

### Acceptance Criteria

- [ ] When the user is about to swap, we show them the link to terms of services below the slider
- [ ] The link and description should match the designs
- [ ] On clicking the link, a web view should open as a modal inside the app
- [ ] [Optional] If the pdf doesn't work inside the web view, we can redirect the user to their native phone browser
- [ ] This webview should load Paraswaps latest terms of services: https://files.paraswap.io/tos_v4.pdf

### Notes

- [ ] Use bottom sheet for WebView (@xAlisher to confirm)
- [ ] Confirm that PDF is visible on both Android and iOs devices

### PR DESCRIPTION
**Pull request by smohamedjavid**
fixes #21012
fixes #21427

### Summary

This PR 
 - Adds a PDF viewer screen to view PDFs within the app instead of opening them on the system browser
 - Updates the swap provider (ParaSwap) terms and conditions to be displayed within the app

| Android | iOS |
| --- | --- |
| <video src="https://github.com/user-attachments/assets/f85cdd3f-cb94-4bd4-81e2-66c2da22850c" /> | <video src="https://github.com/user-attachments/assets/da70af9b-8b59-4038-aaed-7ddd7467b9f2" /> |

#### Platforms

- Android
- iOS

##### Prerequisite: A wallet account with tokens or positive balance

- Open Status
- Navigate to `Wallet > Account > Swap`
- Enter an amount
- Verify on the confirmation page, the link to the Terms and Conditions is shown below the slider
- Verify on tap of **Terms and Conditions**, a screen is opened to show the ParaSwap T&C PDF

status: ready

### CODE REVIEW

File: src/status_im/common/pdf_viewer/view.cljs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+(ns status-im.common.pdf-viewer.view
+  (:require
+    [quo.core :as quo]
+    [react-native.core :as rn]
+    [react-native.pdf-viewer :as pdf-viewer]
+    [utils.re-frame :as rf]))
+
+(defn- on-close
+  []
+  (rf/dispatch [:navigate-back]))
```
Comment (briansztamfater) [inline src/status_im/common/pdf_viewer/view.cljs]: Can use `navigate-back` function from `status-im.common.events-helper`

**Review by briansztamfater — APPROVED** [review]

**Review by shivekkhurana — APPROVED** [review]

File: src/status_im/common/pdf_viewer/view.cljs (hunk 1 above)
Comment (smohamedjavid) [reply src/status_im/common/pdf_viewer/view.cljs]: I was searching for that function in navigation ns 🤦  and ended up adding it to the file. I will update it from `status-im.common.events-helper`. Thanks 🙏

**Review by jakubgs — APPROVED** [review]

**Comment by Horupa-Olena** [comment]
@smohamedjavid Thanks for your PR. 

Could you please let me know if this item " If the pdf doesn't work inside the web view, we can redirect the user to their native phone browser"  from the Acceptance Criteria has been completed? If so, how can it be verified?

**Comment by smohamedjavid** [comment]
> Could you please let me know if this item " If the pdf doesn't work inside the web view, we can redirect the user to their native phone browser" from the Acceptance Criteria has been completed? If so, how can it be verified?

Thank you for testing the PR. 

I believe that's optional and no longer valid, as we use a PDF viewer (instead of a web view) to display the ParaSwap TnC. The objective of the issue is to NOT take the user outside of the app to view the TnC.

cc @shivekkhurana

**Comment by Horupa-Olena** [comment]
@smohamedjavid Thanks for explanation! 

I have no further questions, you can merge this PR.

**Pull request merged**